### ISSUE TITLE
Phoenix request ID mismatches

### ISSUE THREAD

**Issue by jeffkreeftmeijer** [appsignal/appsignal-elixir#282]
From [support](https://app.intercom.io/a/apps/yzor8gyw/respond/inbox/conversation/13340661098). Some Phoenix transactions report the request ID for another request.  It looks like we're taking the request's ID before it's been set by Phoenix, so we're reporting the previous ID somehow.

**Comment by schnittchen** [comment]
I'm trying this out... it looks like git is now required and wasn't previously, is that correct? You might mention that in the README then.

**Comment by jeffkreeftmeijer** [comment]
@schnittchen You can temporarily install the current `develop` branch to test this by switching to git instead of hex to download the AppSignal package, like so. If you test this, please be sure to let us know if the new version fixes this issue for you.

```
{:appsignal, github: "appsignal/appsignal-elixir", branch: "develop"}
```

We'll release a new version of the library (1.4.4) after we've done some testing, and you should be able to switch back to the version released on Hex (`{:appsignal, "~> 1.0"}`) after that. Git is not required to try that release when it's out, only to test the unreleased `develop` branch.

**Comment by schnittchen** [comment]
@jeffkreeftmeijer thank you very much, I took a small sample of three samples and they all match.

**Comment by jeffkreeftmeijer** [comment]
Hey @schnittchen! Thanks a lot for trying it out, and glad to hear the patch fixed the issue for you. We've just released version 1.4.4 of the integration, so you should be able to switch back by using `{:appsignal, "~> 1.0"}` in your dependencies and update the library my running `mix reps.update`.
